shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -642,10 +660,23 @@ public Bitmap getBitmap() {
     private void updateViewportMetrics() {
         if (!isAttached())
             return;
-        mNativeView.getFlutterJNI().setViewportMetrics(mMetrics.devicePixelRatio, mMetrics.physicalWidth,
-                mMetrics.physicalHeight, mMetrics.physicalPaddingTop, mMetrics.physicalPaddingRight,
-                mMetrics.physicalPaddingBottom, mMetrics.physicalPaddingLeft, mMetrics.physicalViewInsetTop,
-                mMetrics.physicalViewInsetRight, mMetrics.physicalViewInsetBottom, mMetrics.physicalViewInsetLeft);
+        mNativeView.getFlutterJNI().setViewportMetrics(
```

Comment:
This is an aside that's out of scope for this PR, but reading this all of these methods that take 20 identical primitives in a very specific order and then handle them in subtle ways seem a little risky. It would be super trivial to accidentally get the order wrong in one of the many places these are called. In Dart this can be sort of addressed trivially with named params, but in Java/C++ there aren't really great options. Could define a struct/class that has these as named members and then pass that around, but that's probably overkill.

Reply:
Right, this crossed my mind while making this change. I had at times passed in the wrong number of params and had to rebuild because there were so many of them and they were all identical. Coming up with a cleaner way to handle this will probably help, especially if we see a world where the number of params continue to grow.